onmt/inputters/item.py

```diff
@@ -0,0 +1,280 @@
+    # Need to add features in last dimensions
+
+    tbatch['src'] = tbatchsrc
+    tbatch['indices'] = torch.LongTensor([ex['indices'] for ex in minibatch])
+    tbatch['srclen'] = torch.LongTensor([len(ex['src']['src_ids'])
+                                         for ex in minibatch])
+
+    if minibatch[0]['tgt'] is not None:
+        tbatchtgt = [torch.LongTensor(ex['tgt']['tgt_ids'])
+                     for ex in minibatch]
+        padidx = vocabs['tgt'][DefaultTokens.PAD]
+        tbatchtgt = pad_sequence(tbatchtgt, padding_value=padidx)
+        tbatchtgt = tbatchtgt[:, :, None]
+        tbatchtgtlen = torch.LongTensor([len(ex['tgt']['tgt_ids'])
+                                         for ex in minibatch])
+        tbatch['tgt'] = tbatchtgt
+        tbatch['tgtlen'] = tbatchtgtlen
+
+    if 'align' in minibatch[0].keys() and minibatch[0]['align'] is not None:
+        sparse_idx = []
+        for i, ex in enumerate(minibatch):
+            for src, tgt in parse_align_idx(ex['align']):
+                sparse_idx.append([i, tgt + 1, src])
+        tbatchalign = torch.LongTensor(sparse_idx)
+        tbatch['align'] = tbatchalign
+
+    if 'src_map' in minibatch[0].keys():
+        src_vocab_size = max([max(ex['src_map']) for ex in minibatch]) + 1
+        src_map = torch.zeros(tbatchsrc.size(0),
+                              len(tbatch['srclen']),
+                              src_vocab_size)
+        for i, ex in enumerate(minibatch):
+            for j, t in enumerate(ex['src_map']):
+                src_map[j, i, t] = 1
+        tbatch['src_map'] = src_map
+
+    if 'alignment' in minibatch[0].keys():
+        alignment = torch.zeros(tbatchtgt.size(0),
+                                len(tbatch['srclen'])).long()
+        for i, ex in enumerate(minibatch):
+            alignment[:len(ex['alignment']), i] = \
+                torch.LongTensor(ex['alignment'])
+        tbatch['alignment'] = alignment
+
+    if 'src_ex_vocab' in minibatch[0].keys():
+        tbatch['src_ex_vocab'] = [ex['src_ex_vocab']
+                                  for ex in minibatch]
+    return tbatch
+
+
+def textbatch_to_tensor(vocabs, batch):
+    """
+    This is a hack to transform a simple batch of texts
+    into a tensored batch to pass through _translate()
+    """
+    numeric = []
+    infer_iter = []
+    for i, ex in enumerate(batch):
+        if isinstance(ex, bytes):
+            ex = ex.decode("utf-8")
+        toks = ex.strip("\n").split()
+        idxs = vocabs['src'](toks)
+        # Need to add features also in 'src'
+        numeric.append({'src': {'src': ex.strip("\n").split(),
+                                'src_ids': idxs},
+                        'srclen': len(ex.strip("\n").split()),
+                        'tgt': None,
+                        'indices': i,
+                        'align': None})
+    numeric.sort(key=text_sort_key, reverse=True)
+    infer_iter = [tensorify(vocabs, numeric)]
+    return infer_iter
```

Comment:
Can't we handle this through a simplified path in `tensorify`?

Reply:
Yes as said in the docstring this is a quick hack to make the translation_server test pass.
The tensorify function requires a very specific json structure, here we need to handle raw text and I think it will require more code to handle features coming from the translation server. I suggest we put htis on hold until we decide for the server.